### PR TITLE
modify upgrade related code to be more defensive

### PR DESCRIPTION
**Pull request by yuval-k**
- always clear the upgrade configs list
- add server side validation
- make sure defaults are applied when settings are nil

### CODE REVIEW

File: projects/gloo/pkg/plugins/utils/upgradeconfig/upgradeconfig_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		err := ValidateRouteUpgradeConfigs(configs)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("should allow websocket upgrade", func() {
+		configs := []*envoyhttp.HttpConnectionManager_UpgradeConfig{{
+			UpgradeType: WebSocketUpgradeType,
+		}}
+		err := ValidateHCMUpgradeConfigs(configs)
+		Expect(err).NotTo(HaveOccurred())
+	})
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/utils/upgradeconfig/upgradeconfig_test.go]: give these tests more descriptive names so they aren't duplicated (route vs hcm)

File: projects/gloo/pkg/plugins/utils/upgradeconfig/upgradeconfig_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+var _ = Describe("Upgradeconfig", func() {
+
+	It("should not error on empty list", func() {
+		err := ValidateHCMUpgradeConfigs(nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("should not error on empty list", func() {
+		err := ValidateRouteUpgradeConfigs(nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/utils/upgradeconfig/upgradeconfig_test.go]: same, more descriptive naming

File: projects/gloo/pkg/plugins/utils/upgradeconfig/upgradeconfig_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	It("should not allow route double websocket upgrade", func() {
+		configs := []*envoyhttp.HttpConnectionManager_UpgradeConfig{{
+			UpgradeType: WebSocketUpgradeType,
+		}, {
+			UpgradeType: WebSocketUpgradeType,
+		}}
+		err := ValidateHCMUpgradeConfigs(configs)
+		Expect(err).To(HaveOccurred())
+	})
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/utils/upgradeconfig/upgradeconfig_test.go]: naming

**Comment by yuval-k** [comment]
@kdorosh  to reduce redundancy I added context to the tests.

**Pull request merged**